flocker/control/test/test_schemas.py

```diff
@@ -98,6 +98,41 @@
                 {'internal': 80, 'external': 8080},
             ]
         },
+        # Environment given but not a dict
+        {
+            'host': '192.168.0.3',
+            'image': 'postgres',
+            'name': 'postgres',
+            'environment': 'x=y'
+        },
+        # Environment given but at least one entry is not a string
+        {
+            'host': '192.168.0.3',
+            'image': 'postgres',
+            'name': 'postgres',
+            'environment': {
+                'POSTGRES_USER': 'admin',
+                'POSTGRES_VERSION': 9.4
+            }
+        },
+        # Environment given but a key is too long (>4096 characters)
+        {
+            'host': '192.168.0.3',
+            'image': 'postgres',
+            'name': 'postgres',
+            'environment': {
+                'MY_'+('X'*4094): 'somevalue',
```

Comment:
whitespace around operators.  Also, this construct seems needlessly complicated.  Why not just `'X' * 4097`?